src/App.tsx

```diff
@@ -1,24 +1,202 @@
 
-const USER_ID = 0;
+const USER_ID = 6101;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState('all');
+  const [error, setError] = useState('');
+  const [isItError, setIsItError] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((data) => {
+        setTodos(data);
+        setError('');
+      })
+      .catch(() => {
+        setIsItError(true);
+        setError(Errors.LOADING);
+      })
+      .finally(() => {
+        setTimeout(() => {
+          setIsItError(false);
+        }, 3000);
+      });
+  }, []);
+
+  const handleAddTodo = (todo: Todo) => {
+    if (!todo.title) {
+      setError(Errors.TITLE);
+      setIsItError(true);
+
+      setTimeout(() => {
+        setIsItError(false);
+      }, 3000);
+
+      return;
+    }
+
+    setTempTodo(todo);
+
+    postTodo(todo)
+      .then(() => (
+        setTimeout(() => {
+          setTodos([...todos, todo]);
+          setTempTodo(null);
+        }, 1000)
+      ))
+      .catch(() => {
+        setError(Errors.ADDING);
+        setIsItError(true);
+
+        setTimeout(() => {
+          setIsItError(false);
+        }, 3000);
+      });
+  };
+
+  const handleRemoveTodo = (todoId: number) => {
+    removeTodo(todoId)
+      .then(() => {
+        const filteredTodos = todos.filter((todo) => todo.id !== todoId);
+
+        setTodos(filteredTodos);
+      })
+      .catch(() => {
+        setError(Errors.REMOVING);
+        setIsItError(true);
+
+        setTimeout(() => {
+          setIsItError(false);
+        }, 3000);
+      });
+  };
+
+  const handleClearTodos = async (todoIds: number[]) => {
+    try {
+      await Promise.all(todoIds.map((id) => handleRemoveTodo(id)));
+    } catch (mistake) {
+      setError(Errors.REMOVING);
+    }
+  };
+
+  const visibleTodos = todos
+    .filter((todo) => {
+      switch (filter) {
+        case Filter.ACTIVE:
+          return !todo.completed;
+        case Filter.COMPLETED:
+          return todo.completed;
+        default:
+          return true;
+      }
+    });
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          <button
+            type="button"
+            className={classNames(
+              'todoapp__toggle-all',
+              { active: todos.some((todo) => !todo.completed) },
```

Comment:
i'd keep it active all-the-time
so you can flip all tasks even if all are completed,
but not crucial

Reply:
Done!